### DISCUSSION TITLE
Sending message to specific user

### DISCUSSION THREAD

**Discussion by yohami** [colyseus/colyseus#784] [Q&A]
We're using Colyseus for a social gaming app where a user can be connected on several devices (say, desktop browser and phone).

Need to send a direct message to a user, to all of their logged devices.

We're migrating from socket.io and this is how we had it set up:

On login, the user connects to a room that is named with their userID. To send a direct message to that user, we simply emit a message to that room. This means the user can be connected from several devices to that room, and they'll get anything that is sent there.

Is there a way to do this in Colyseus?

Is it possible to create a room with a specific ID and then sending messages to it from the server?

I guess we can make a database and keep ids of rooms etc then having the server join them to deliver messages and then logging out but there's probably a better way.

Any help appreciated, thanks

Yohami

**Comment by yohami** [comment]
I found it can be done with the Presence API using redis pub/sub

**Comment by endel** [comment]
Hi @yohami, sorry for the delay to answer - you're correct, you can use both the Presence API, and [`matchMaker.remoteRoomCall(roomId, ...)`](https://docs.colyseus.io/server/matchmaker/#remoteroomcallroomid-method-args) to communicate between rooms if you know their `roomId`
